css/src/components/progress-bar.scss

```diff
@@ -0,0 +1,99 @@
+$progress-bar-height-medium: 2px;
+$progress-bar-height-large: 4px;
+
+$progress-bar-background: $secondary-background !default;
+$progress-bar-color: $primary !default;
+$progress-bar-success-color: $success !default;
+$progress-bar-warning-color: $warning !default;
+$progress-bar-danger-color: $danger !default;
+$progress-bar-border-radius-rounded: $border-radius-rounded !default;
+
+$progress-bar-animation-duration: 2s;
+
+.progress-bar-container {
+	position: relative;
+	width: 100%;
+	border-radius: $progress-bar-border-radius-rounded;
+	background-color: $progress-bar-background;
+	overflow: hidden;
+
+	.theme-high-contrast & {
+		background-color: $secondary-background-glow-high-contrast;
+	}
+
+	&.progress-bar-md {
+		height: $progress-bar-height-medium;
+	}
+
+	&.progress-bar-lg {
+		height: $progress-bar-height-large;
+	}
+
+	&.progress-bar-square {
+		border-radius: 0;
+	}
+
+	&.progress-bar-rounded {
+		border-radius: $progress-bar-border-radius-rounded;
+	}
+}
+
+@mixin indeterminate-gradient($color) {
+	position: absolute;
+	width: 100%;
+	animation: indeterminate-animation $progress-bar-animation-duration infinite linear;
+	background: linear-gradient(to right, transparent 0%, $color 50%, transparent 100%);
+	inset-block-start: 0;
+	inset-inline-start: 0;
+}
+
+.progress-bar {
+	height: 100%;
+	transition: width 0.3s ease;
+	background-color: $progress-bar-color;
+
+	&.progress-bar-success {
+		background-color: $progress-bar-success-color;
+
+		&.indeterminate {
+			@include indeterminate-gradient($progress-bar-success-color);
+		}
+	}
+
+	&.progress-bar-warning {
+		background-color: $progress-bar-warning-color;
+
+		&.indeterminate {
+			@include indeterminate-gradient($progress-bar-warning-color);
+		}
+	}
+
+	&.progress-bar-danger {
+		background-color: $progress-bar-danger-color;
+
+		&.indeterminate {
+			@include indeterminate-gradient($progress-bar-danger-color);
+		}
+	}
+
+	&.indeterminate {
+		@include indeterminate-gradient($progress-bar-color);
+	}
+}
+
+@keyframes indeterminate-animation {
```

Comment:
this should go to `animations.scss` file

Reply:
And we may also want to name it something specifically corresponding to progress bars -- maybe `progress-bar-indeterminate-animation`?